libpod/image/image.go

```diff
@@ -410,48 +410,92 @@ func (ir *Runtime) getLocalImage(inputName string) (string, *storage.Image, erro
 	if inputName == "" {
 		return "", nil, errors.Errorf("input name is blank")
 	}
+
 	// Check if the input name has a transport and if so strip it
 	dest, err := alltransports.ParseImageName(inputName)
 	if err == nil && dest.DockerReference() != nil {
 		inputName = dest.DockerReference().String()
 	}
 
-	img, err := ir.getImage(stripSha256(inputName))
```

Comment:
This removes the only caller of `getImage` with a name AFAICS; the remaining  _know_ that they are using a full image ID; so, `getImage` should become `getImageByID` and use either `NewStoreReference(store, nil, id)` (more explicit about the intent) or perhaps `store.Image` (bypasses c/image, but allows names as input), without the overhead/risk of `ParseStoreReference` allowing other inputs.

Reply:
Very good catch. I can open a follow-up PR to clean up :+1: